docs/src/main/sphinx/installation/kubernetes.rst

```diff
@@ -0,0 +1,342 @@
+      cluster with the ``kind-`` prefix added.
+
+   You should see similar output with a random port assigned if you are
+   running on kind.
+
+   .. code-block:: text
+
+      Kubernetes control plane is running at https://127.0.0.1:59553
+      CoreDNS is running at https://127.0.0.1:59553/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+
+#. (Optional) Create ``values.yaml`` file to define the parameters of your
+   deployment.
+
+   The helm chart has a `set of default values <https://github.com/trinodb/charts/blob/main/charts/trino/values.yaml>`_
+   that are used if you don't specify your own ``values.yaml`` file. When you
+   create this file, you will override only the values you specify. The
+   remaining properties will remain as the default. Add a ``values.yaml``
+   with the following configuration:
+
+   .. code-block:: yaml
+
+      image:
+        tag: "|trino_version|"
+      server:
+        workers: 2
+        config:
+          memory:
+            heapHeadroomPerNode: "1GB"
+          query:
+            maxMemory: "4GB"
+            maxMemoryPerNode: "1GB"
+      coordinator:
+        jvm:
+          maxHeapSize: "8G"
+      worker:
+        jvm:
+          maxHeapSize: "8G"
+
+   Let's explain what these values do and some useful notes.
+
+   #. ``.Values.image.tag`` is set to the current version, |trino_version|. Set
+      this value if you need to use a specific version of Trino. The default is
+      ``latest``.
+   #. ``.Values.server.workers`` is set to ``2``. This value sets the number of
+      workers, in this case, a coordinator and two worker nodes will get
+      deployed.
+   #. ``.Values.server.config.memory.heapHeadroomPerNode`` is set to ``1GB``.
+      This sets the :ref:`prop-resource-memory-heap-headroom-per-node` property.
+   #. ``.Values.server.config.query.maxMemory`` is set to ``4GB``.
+      This sets the :ref:`prop-resource-query-max-memory` property.
+   #. ``.Values.server.config.query.maxMemoryPerNode`` is set to ``1GB``.
+      This sets the :ref:`prop-resource-query-max-memory-per-node` property.
+   #. ``.Values.coordinator.jvm.maxHeapSize`` is set to ``8GB``.
+      This sets the ``Xmx`` property in the :ref:`jvm_config` on the
+      coordinator.
+   #. ``.Values.worker.jvm.maxHeapSize`` is set to ``8GB``.
+      This sets the ``Xmx`` property in the :ref:`jvm_config` on the
+      worker.
+
+      .. warning::
+         When updating memory configurations, you need to make sure that
+         :ref:`prop-resource-query-max-memory` is greater than
+         :ref:`prop-resource-query-max-memory-per-node`. Also make sure that the
+         sum of :ref:`prop-resource-query-max-memory-per-node` and
+         :ref:`prop-resource-memory-heap-headroom-per-node` is less than the
+         ``Xmx`` property in the :ref:`jvm_config`.
+
+
+#. Add the Trino Helm chart repository to Helm if you haven't done so already.
```

Comment:
Shouldn't this be the first step? It only has to be done once.

Reply:
Yeah, I suppose it could be. I think maybe validating kubectl is also a good first step. This can definitely be the second. It forces you to validate your tools are working before getting started with any configs.

Reply:
Either order is fine .. you could even argue that you these are prerequisites.. but thats splitting hairs.. good as it imho